store/rootmulti/store.go

```diff
@@ -407,16 +413,19 @@ func (rs *Store) Commit() types.CommitID {
 		// a 'snapshot' height.
 		if rs.pruningOpts.KeepEvery == 0 || pruneHeight%int64(rs.pruningOpts.KeepEvery) != 0 {
 			rs.pruneHeights = append(rs.pruneHeights, pruneHeight)
+			rs.flushPruningHeights()
 		}
 	}
 
 	// batch prune if the current height is a pruning interval height
 	if rs.pruningOpts.Interval > 0 && version%int64(rs.pruningOpts.Interval) == 0 {
 		rs.logger.Info("pruning", "height", version, "to_prune", rs.pruneHeights)
-		pruneErr = rs.pruneStores()
+		rs.pruneStores()
 	}
 
+	rs.mx.RLock()
```

Comment:
This is probably redundant since we always write `lastCommitInfo` during `Commit()` but still added for safety